SnakeGame/Snake.js

```diff
@@ -0,0 +1,207 @@
+class Snake {
+    constructor(game) {
+        this.game = game;
+        this.direction = "R";
+        this.squeue = new Queue();
+        this.head = [2, 3];
+        // start off as a 3 unit long snake
+        this.squeue.enqueue([2, 1]);
+        this.squeue.enqueue([2, 2]);
+        this.squeue.enqueue([2, 3]);
+        this.appleCount = 0;
+        this.input = "R";
+
+        pageCounter.textContent = "Apples Eaten: " + this.appleCount;
+    }
+
+    move() {
+        let dir = this.direction;
+        if (this.input == "R" && dir == "L"
+            || this.input == "L" && dir == "R"
+            || this.input == "U" && dir == "D"
+            || this.input == "D" && dir == "U") {
+        } else {
+            this.direction = this.input;
+        }
+        dir = this.direction;
```

Comment:
This section of code (from the start of the function to this line) feels a little messy/inconsistent. It feels weird to have to define `dir` twice, and not have an alias for `this.input` as you do with `this.direction`. (Saving a little typing is OK but not really necessary here.) If you want to remove the second `dir = this.direction` (line 26), you can change line 24 to be:
```javascript
dir = this.direction = this.input;
```